src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -807,9 +809,10 @@ namespace cryptonote
         return 1;
       }
       //to avoid concurrency in core between connections, suspend connections which delivered block later then first one
+      const crypto::hash block_hash = get_block_hash(b);
       if(count == 2)
       {
-        if(m_core.have_block(get_block_hash(b)))
+        if(m_core.have_block(block_hash))
```

Comment:
The comment that was added with this block of code:

> to avoid concurrency in core between connections, suspend connections which delivered block later then first one

So if the comment is accurate, the goal was similar to that of this patch. I _think_ the core is only checked once to avoid locking. It would be nice to know why `count == 2` was chosen as the filtering criteria.

Unfortunately it seems that your level of confidence in changing this portion of the code is low - definitely can't fault you here. Moving this to `if (empty() || m_core.have_block(arg.blocks.back()) return;` before the loop seems like it would be a nice bump in quickly filtering duplicate information, so just something for everyone to keep in mind.